### PR TITLE
doc(HMS-3527): add rbac dev documentation

### PR DESCRIPTION
**Pull request by avisiedo**
Update developer documentation adding a new rbac section to the development documentation.

### CODE REVIEW

File: docs/dev/08-rbac.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+to let us check the operations locally. This component exists
+for allowing automation on the smoke tests and future integration
+tests, so we can check different scenarios as we need.
+
+The rbac mock load 4 different profiles, so we can choose when
+the service is started which profile to use by setting the
+`APP_CLIENTS_RBAC_PROFILE` variable. The profiles that we can
+use are:
+- `super-admin`: only contain `*:*:*`
+- `domain-admin`: contain the default Domain Admin permissions.
```
Comment (tiran) [inline docs/dev/08-rbac.md:13]:
```suggestion
- `domain-admin` (default): contain the default Domain Admin permissions.
```

File: docs/dev/08-rbac.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+- Stop the rbac mock by: `make mock-rbac-down`
+
+By default, the profile that is used by the rbac mock
+is `domain-admin`; but you can use `super-admin`,
+`domain-admin`, `domain-readonly`, `domain-no-perms`
+or `custom`.
+
+```sh
+$ make mock-rbac-up APP_CLIENTS_RBAC_PROFILE
+```
```
Comment (tiran) [inline docs/dev/08-rbac.md:33]:
```suggestion
- Start with a different profile: `make mock-rbac-up APP_CLIENTS_RBAC_PROFILE=domain-readonly`
```

**Review by tiran — CHANGES_REQUESTED** [review]
Could you please explain briefly, how each permission influences idmsvc-backend. How does access as `super-admin` behave differently than `domain-admin`?

**Comment by avisiedo** [comment]
> Could you please explain briefly, how each permission influences idmsvc-backend. How does access as `super-admin` behave differently than `domain-admin`?

Sure! `super-admin` is using wildcards, meanwhile `super-admin` is providing the whole set of permissions that our service manage.

In terms of behavior of our service, both profiles must behave in the same way! the existence of the `super-admin` profile is to have evidences on our tests (or some bug replayed on the future) that our service is managing correctly the wildcards when checking a required permissions against the ACL list returned from the rbac microservice.

**Comment by avisiedo** [comment]
> Could you please explain briefly, how each permission influences idmsvc-backend. How does access as `super-admin` behave differently than `domain-admin`?

Sorry! I thought about this as a clarification on the comments of the PR instead of the content of the documentation. I have updated the content.

**Review by frasertweedale — APPROVED** [review]

**Pull request merged**